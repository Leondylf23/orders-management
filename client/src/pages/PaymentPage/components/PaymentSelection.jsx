import { useDispatch, connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { FormattedMessage } from 'react-intl';
import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import AccountBalanceIcon from '@mui/icons-material/AccountBalance';
import PointOfSaleIcon from '@mui/icons-material/PointOfSale';

import { selectUserInputData } from '../selectors';
import { setUserInputs } from '../actions';

import classes from '../style.module.scss';

const PaymentSelectionComponent = ({ inputtedData }) => {
  const dispatch = useDispatch();

  const [selectedPayment, setSelectedPayment] = useState(null);

  const selectPaymentBtn = (isTransfer) => {
    const temp = isTransfer
      ? {
          id: 'TRANSFER',
          nameIntlId: 'payment_payment_transfer',
        }
      : {
          id: 'CASH',
          nameIntlId: 'payment_payment_onsitepay',
        };

    setSelectedPayment(temp);
    dispatch(setUserInputs({ ...inputtedData, paymentMethod: temp }));
  };

  useEffect(() => {
    if (inputtedData?.paymentMethod) {
      setSelectedPayment(inputtedData?.paymentMethod);
    }
  }, [inputtedData]);

  return (
    <div className={classes.componentContainer} data-testid="payment-page-payment-selection">
      <h2 className={classes.title}>
        <FormattedMessage id="payment_step_3_name" />
      </h2>
      <h4 className={classes.pageTitle}>
        <FormattedMessage id="payment_payment_page_title" />
      </h4>
      <div className={classes.paymentSelectionContainer}>
        <div
          className={classes.payment}
          onClick={() => selectPaymentBtn(true)}
          data-active={selectedPayment?.id === 'TRANSFER'}
        >
          <AccountBalanceIcon />
          <p className={classes.text}>
            <FormattedMessage id="payment_payment_transfer" />
          </p>
        </div>
        <div
          className={classes.payment}
          onClick={() => selectPaymentBtn(false)}
          data-active={selectedPayment?.id === 'CASH'}
        >
          <PointOfSaleIcon />
          <p className={classes.text}>
            <FormattedMessage id="payment_payment_onsitepay" />
          </p>
        </div>
      </div>
    </div>
  );
};

PaymentSelectionComponent.propTypes = {
  inputtedData: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  inputtedData: selectUserInputData,
});

export default connect(mapStateToProps)(PaymentSelectionComponent);
