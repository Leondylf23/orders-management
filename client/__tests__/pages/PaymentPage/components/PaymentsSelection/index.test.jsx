import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
// import RouterDom from 'react-router-dom';

import PaymentSelection from '@pages/PaymentPage/components/PaymentSelection';
import store from '@store';
import Language from '@containers/Language';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: jest.fn(),
}));

jest.mock('reselect', () => ({
  ...jest.requireActual('reselect'),
}));

const ParentComponent = (children) => (
  <Provider store={store}>
    <Language>{children}</Language>
  </Provider>
);

// const navigate = jest.fn().mockName('navigate');

describe('Payment Selection', () => {
  // beforeEach(() => {
  //     jest.spyOn(RouterDom, 'useNavigate').mockImplementation(() => navigate);
  // });

  test('Rendered', () => {
    const { getByTestId } = render(ParentComponent(<PaymentSelection />));
    const paymentSelection = getByTestId('payment-page-payment-selection');
    expect(paymentSelection).toBeInTheDocument();
  });

  test('Should match with snapshot', () => {
    const paymentSelection = render(ParentComponent(<PaymentSelection />));
    expect(paymentSelection).toMatchSnapshot();
  });
});
