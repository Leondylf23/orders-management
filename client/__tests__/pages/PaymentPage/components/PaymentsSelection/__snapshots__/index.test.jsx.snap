// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Selection Should match with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="componentContainer"
        data-testid="payment-page-payment-selection"
      >
        <h2
          class="title"
        >
          Pembayaran
        </h2>
        <h4
          class="pageTitle"
        >
          Pilih metode pembayaran Anda
        </h4>
        <div
          class="paymentSelectionContainer"
        >
          <div
            class="payment"
            data-active="false"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="AccountBalanceIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M4 10h3v7H4zm6.5 0h3v7h-3zM2 19h20v3H2zm15-9h3v7h-3zm-5-9L2 6v2h20V6z"
              />
            </svg>
            <p
              class="text"
            >
              Transfer
            </p>
          </div>
          <div
            class="payment"
            data-active="false"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="PointOfSaleIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17 2H7c-1.1 0-2 .9-2 2v2c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 4H7V4h10v2zm3 16H4c-1.1 0-2-.9-2-2v-1h20v1c0 1.1-.9 2-2 2zm-1.47-11.81C18.21 9.47 17.49 9 16.7 9H7.3c-.79 0-1.51.47-1.83 1.19L2 18h20l-3.47-7.81zM9.5 16h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm3 4h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm3 4h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5z"
              />
            </svg>
            <p
              class="text"
            >
              Bayar di tempat
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="componentContainer"
      data-testid="payment-page-payment-selection"
    >
      <h2
        class="title"
      >
        Pembayaran
      </h2>
      <h4
        class="pageTitle"
      >
        Pilih metode pembayaran Anda
      </h4>
      <div
        class="paymentSelectionContainer"
      >
        <div
          class="payment"
          data-active="false"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AccountBalanceIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M4 10h3v7H4zm6.5 0h3v7h-3zM2 19h20v3H2zm15-9h3v7h-3zm-5-9L2 6v2h20V6z"
            />
          </svg>
          <p
            class="text"
          >
            Transfer
          </p>
        </div>
        <div
          class="payment"
          data-active="false"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="PointOfSaleIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M17 2H7c-1.1 0-2 .9-2 2v2c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm0 4H7V4h10v2zm3 16H4c-1.1 0-2-.9-2-2v-1h20v1c0 1.1-.9 2-2 2zm-1.47-11.81C18.21 9.47 17.49 9 16.7 9H7.3c-.79 0-1.51.47-1.83 1.19L2 18h20l-3.47-7.81zM9.5 16h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm3 4h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm3 4h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5zm0-2h-1c-.28 0-.5-.22-.5-.5s.22-.5.5-.5h1c.28 0 .5.22.5.5s-.22.5-.5.5z"
            />
          </svg>
          <p
            class="text"
          >
            Bayar di tempat
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
